[Toh Sihui] iP

tsh22 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

tsh22 committed:
Added greet, echo and exit functions of Duke

tsh22 committed:
Replaced echo function with add and list functions

tsh22 committed:
Added new Task.java class

tsh22 committed:
Added mark task as done function

tsh22 committed:
Modified Task.java's toString() method

tsh22 committed:
Modified print statements using new toString() method in 'Task.java' for improved clarity

tsh22 committed:
Added Deadline.java, Event.java, Todo.java classes

tsh22 committed:
Added support for tracking three types of tasks

tsh22 committed:
Updating input.txt and EXPECTED.TXT files for test automation

tsh22 committed:
Updating toString() methods of Deadline.java, Event.java to correct differences during automated testing

tsh22 committed:
Modified nested if statements to switch cases for improved readability

tsh22 committed:
Modified input.txt, EXPECTED.TXT to include test cases for error handling level-5

tsh22 committed:
Added error handling for if user inputs invalid commands or does not specify task description

tsh22 committed:
Modified input.txt, EXPECTED.TXT to include test cases for delete function level-6

tsh22 committed:
Added remove function in Duke.java

tsh22 committed:
Reformat code to match code convention

tsh22 committed:
Add TaskList.java

Improve encapsulation of taskList behaviour

Let's
* Add addTask method
* Add deleteTask method
* Add taskDone method
* Add getListSize method
* Add printList method

tsh22 committed:
Replace 'currList' variable with taskList and implement relevant methods

tsh22 committed:
Add new attribute 'taskType' to Task object

jlxw48 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,88 @@
+                item.markAsDone();
+                System.out.println("Nice job! I've marked this task as done:\n"
+                        + item.toString());
+            } else if ((commandArr[0]).equals("delete")){
```
Do you think it will be better if you insert a whitespace between ')' and '{'?

jlxw48 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,88 @@
+                String wholeTask;
+                Task item;
+                switch (taskType) {
+                    case "todo":
```
Do you think it will be more organised if the case blocks are indented at the same level as the switch block?

jlxw48 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,88 @@
+import java.util.*;
```
Do you think it would be better to import only the necessary libraries?

jlxw48 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+    public String toString() {
+        return "[D]" + super.toString() + "(" + by + ")";
+    }
+}
```
Remember to include newlines at the end of each file!

jlxw48 reviewed (commented):
Generally looks good! 
Minimal coding style errors.

tsh22 committed:
Add Storage.java

Require new functionality of saving task list to hard disk.

Add methods within Storage class to support functionality of saving task list to hard disk.

tsh22 committed:
Add TaskList.java

Duke class is handling too many details relating to the tasklist.

Add TaskList class to enable better abstraction of functions relating to the taskList.

tsh22 committed:
Modify Deadline.java, Event.java and Task.java to support functions in Storage.java

tsh22 committed:
Modify Duke.java to implement Storage.class functionality

tsh22 committed:
Add data folder with duke.txt file that stores task list in hard drive

tsh22 committed:
Add data folder with duke.txt file that stores task list in hard drive

tsh22 committed:
Merge branch 'branch-Level-7'

tsh22 committed:
Change attribute type so that Deadline and Event tasks can understand dates

tsh22 committed:
Change how user input is split under 'Deadline' and 'Event' cases

tsh22 committed:
Merge pull request #1 from tsh22/branch-Level-8

Merge branch 'branch-Level-8'

tsh22 committed:
Add Parser.java and UI.java

tsh22 committed:
Add Command.java

tsh22 committed:
Add DeadlineCommand.java, EventCommand.java, TodoCommand.java that inherit from Command.java abstract class

tsh22 committed:
Add DeleteCommand.java, DoneCommand.java, ExitCommand.java, ListCommand.java that inherit from Command.java

tsh22 committed:
Add InvalidTaskCommand.java that inherits from abstract class Command.java

tsh22 committed:
Extract code from TaskList.java and Duke.java to be handled by Ui.java and Parser.java instead

tsh22 committed:
Modify visibility of attributes in Deadline.java, Event.java, Task.java

tsh22 committed:
Add Task related classes into 'Tasks' folder

tsh22 committed:
Add Command related classes to 'Commands' folder

tsh22 committed:
Extract java files from Command and Tasks folders

tsh22 committed:
Remove commented out code

jared98lyj commented on src/main/java/Duke.java:
```diff
@@ -35,9 +35,14 @@ public static void main(String[] args) {
+                Task item = currList.remove(Integer.parseInt(commandArr[1]) - 1);
+                System.out.println("Gotcha. I've removed this task:\n"
+                        + item.toString()
+                        + "\nNow you have " + currList.size() + " task(s) in the list.");
```
Shift the lines to make it top down (42 to 41)?

jared98lyj commented:
Generally good code structure. Good use of inheritance for the task class. Code is also neat without unecessary repeats. Some general formatting issues, but are minor.

tsh22 committed:
Add DukeException.java class

tsh22 committed:
Modify code to support exception handling via DukeException class

tsh22 committed:
Add JUnit tests under new DukeTest.java file

tsh22 committed:
Modify exception handling for EventCommand and DeadlineCommand

tsh22 committed:
Modify Ui.java to print correct message to user

tsh22 committed:
Create JAR file

tsh22 committed:
Add FindCommand.java

tsh22 committed:
Add methods in Parser.java, TaskList.java and Ui.java to support find command functionality

tsh22 committed:
Add JavaDoc comments to all classes and methods

tsh22 committed:
Modify code to adhere to coding standards

tsh22 committed:
Merge branch 'branch-Level-9'

tsh22 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/TaskList.java

Resolution: Remove getListSize() method from TaskList.java

tsh22 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/TaskList.java

Resolution: Rename taskDone method as markTaskDone to adhere to coding convention.

tsh22 committed:
Merge branch 'master' into branch-Level-9

tsh22 committed:
Merge branch 'master' into branch-A-JavaDoc

tsh22 committed:
Merge branch 'master' into branch-A-CodingStandard

tsh22 committed:
Add JavaDoc comments to FindCommand class and methods and findTasks method of TaskList class.

tsh22 committed:
Merge branch 'branch-A-JavaDoc'

tsh22 committed:
Merge branch 'branch-Level-9'

tsh22 committed:
Merge branch 'branch-A-CodingStandard'

tsh22 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tsh22 committed:
Set up gradle

tsh22 committed:
Move all files into 'duke' package

tsh22 committed:
Merge branch 'master' into branch-A-Gradle

tsh22 committed:
Modify mainClassName to allow gradle to run

tsh22 committed:
Merge branch 'branch-A-Gradle'

tsh22 committed:
Modify gradle mainClassName for duke to run

tsh22 committed:
Create GUI for Duke

tsh22 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	build.gradle

Resolution: Replace mainClassName in gradle to duke.Launcher

tsh22 committed:
Modify JavaDocs to adhere to CodingStandards

tsh22 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/Ui.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/ExitCommand.java

Resolution: Delete outdated JavaDocs

tsh22 committed:
Add checkstyle.xml to use checkstyle-IDEA plugin

tsh22 committed:
Modify code according to coding standards

tsh22 committed:
Add functionality for user to update tasks

tsh22 committed:
Modify JavaDoc of methods

tsh22 committed:
Add support in Storage class for creating a directory if nonexistent

tsh22 committed:
Modify exception handling

tsh22 committed:
Modify JavaDoc

tsh22 committed:
Merge branch 'master' into branch-A-Assertions

tsh22 committed:
Add assert statements at various points of code

tsh22 committed:
Merge branch 'master' into branch-A-CodeQuality

tsh22 committed:
Refactor code to improve code quality

tsh22 committed:
Merge pull request #4 from tsh22/branch-A-CodeQuality

PR branch-A-CodeQuality

tsh22 committed:
Merge pull request #3 from tsh22/branch-A-Assertions

PR branch-A-Assertions

tsh22 committed:
Improve GUI

tsh22 committed:
Modify README.md with user guide

tsh22 committed:
Gave credit

tsh22 committed:
Set theme jekyll-theme-hacker

tsh22 committed:
Modify code to comply with coding standards

tsh22 committed:
Merge branch 'master' of https://github.com/tsh22/ip

tsh22 committed:
Edit README.md

tsh22 committed:
Set theme jekyll-theme-cayman

tsh22 committed:
Modify README.md

tsh22 committed:
Edit link to Duke screenshot in README.md

tsh22 committed:
Update README.md

tsh22 committed:
Update README.md

tsh22 committed:
Update README.md

tsh22 committed:
Merge branch 'master' of https://github.com/tsh22/ip
